### PR TITLE
Remove `suffix` from `MetaItemLit`

### PR DESCRIPTION
**Pull request by camsteffen**
...as well as `StrLit` for which I believe it is always `None`. Otherwise, I believe the only functional change is in ast pretty printing where `1_u32` may change to `1u32`.

### CODE REVIEW

**Comment by camsteffen** [comment]
I found a couple usages in rustfmt where unfortunately this may change behavior for a very specific case - formatting hex literals. I tried to verify this with a rustfmt test but `./x test src/tools/rustfmt` doesn't seem to run the source/target tests?

**Comment by Centri3** [comment]
> I found a couple usages in rustfmt where unfortunately this may change behavior for a very specific case - formatting hex literals. I tried to verify this with a rustfmt test but `./x test src/tools/rustfmt` doesn't seem to run the source/target tests?
> 
> Edit: Oh well I'll just commit the test.

clippy parses the suffix manually in cases where it's necessary from the literal's span. Not sure if rustfmt can/should do this tho - just the first thing that came to mind

File: compiler/rustc_ast/src/ast.rs [hunk 1]
```diff
@@ -1791,7 +1787,7 @@ impl StrLit {
             StrStyle::Cooked => token::Str,
             StrStyle::Raw(n) => token::StrRaw(n),
         };
-        token::Lit::new(token_kind, self.symbol, self.suffix)
+        token::Lit::new(token_kind, self.symbol, None)
```
[the remark this answers is not in the document]
Comment (davidtwco) [reply compiler/rustc_ast/src/ast.rs:1790]: Both are used in macros.

**Comment by calebcartwright** [comment]
> I tried to verify this with a rustfmt test but ./x test src/tools/rustfmt doesn't seem to run the source/target tests?

It definitely should and has previously in my experience. If you want to confirm locally then doing something like adding a comment within any of the "before" files under `source` should trigger a failure

**Comment by Dylan-DPC** [comment]
@camsteffen any updates on this? thanks

**Comment by oskgo** [comment]
@camsteffen

Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this.
Note: if you are going to continue please open the PR BEFORE you push to it, else you won't be able to reopen - this is a quirk of github.
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**